### PR TITLE
114-send-emails-to-supplier-when-its-been-withdrawn

### PR DESCRIPTION
**Pull request by benvand**
https://trello.com/c/WXIYqtgX/114-send-emails-to-supplier-when-its-been-withdrawn

notify_suppliers_of_brief_withdrawals.yml 
Job running daily at 8am
See:
https://github.com/alphagov/digitalmarketplace-scripts/pull/171

### CODE REVIEW

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    triggers:
+      - timed: "H 8 * * *"
+    publishers:
+      - trigger-parameterized-builds:
+          - project: notify-slack
+            condition: UNSTABLE_OR_WORSE
+            predefined-parameters: |
+              USERNAME=brief-withdrawals
+              JOB=Notify suppliers of brief withdrawals {{ environment }}
+              ICON=:question:
```
Comment (katstevens) [inline job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Do you want to pick a different icon to the 'notify suppliers of new questions and answers' job?

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+          if [ -n "$BRIEF_ID" ]; then
+            FLAGS="$FLAGS --brief_id=$BRIEF_ID"
+          fi
+
+          if [ "$DRY_RUN" = "true" ]; then
+            FLAGS="$FLAGS --dry-run"
+          fi
+
+          docker run --user $(id -u) digitalmarketplace/scripts scripts/notify-suppliers-of-brief-withdrawal.py '{{ environment }}' "$DM_DATA_API_TOKEN_{{ environment|upper }}" $NOTIFY_API_TOKEN 3e43a131-8399-4674-89d3-b4f80d2e07d4 $FLAGS
```
Comment (katstevens) [inline job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Do we need the `--user $(id -u)` arg here? I thought this was only if we needed local write permissions?

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+              STATUS=FAILED
+              URL=<${BUILD_URL}consoleFull|#${BUILD_NUMBER}>
+              CHANNEL=#dm-release
+    builders:
+      - shell: |
+
+          FLAGS=""
+
+          if [ -n "$WITHDRAWN_DATE" ]; then
+            FLAGS="$FLAGS --withdrawn_date=$WITHDRAWN_DATE"
```
Comment (katstevens) [inline job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Not sure about having the `FLAGS=""` above - is there a reason you've not just done `FLAGS="--withdrawn_date=$WITHDRAWN_DATE" on this line instead?

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{% set environments = ['preview', 'staging', 'production'] %}
+---
+{% for environment in environments %}
+- job:
+    name: "notify-suppliers-of-brief_withdrawals-{{ environment }}"
```
[the remark this answers is not in the document]
Comment (benvand) [reply job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Done

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml (hunk 1 above)
Comment (benvand) [reply job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Done

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml (hunk 3 above)
Comment (benvand) [reply job_definitions/notify_suppliers_of_brief_withdrawals.yml]: The fear of the undefined variable 👻  🎃 
Changed

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml (hunk 2 above)
Comment (benvand) [reply job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Maybe not but I still need to test this on jenkins, will see if it works without, have changed for now

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{% set environments = ['preview', 'staging', 'production'] %}
```
[the remark this answers is not in the document]
Comment (benvand) [reply job_definitions/notify_suppliers_of_brief_withdrawals.yml]: I'm pretty easy about this, I want to see what happens with it on preview and staging. I'll give it a test when I deploy and if not I'll strip it back off. I think staging should work though. As long as it has the notify api key

File: job_definitions/notify_suppliers_of_brief_withdrawals.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+          if [ -n "$BRIEF_ID" ]; then
+            FLAGS="$FLAGS --brief_id=$BRIEF_ID"
+          fi
+
+          if [ "$DRY_RUN" = "true" ]; then
+            FLAGS="$FLAGS --dry-run"
+          fi
+
+          docker run digitalmarketplace/scripts scripts/notify-suppliers-of-brief-withdrawal.py '{{ environment }}' "$DM_DATA_API_TOKEN_{{ environment|upper }}" $NOTIFY_API_TOKEN 3e43a131-8399-4674-89d3-b4f80d2e07d4 $FLAGS
```
Comment (Wynndow) [inline job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Should this api key be exposed like this?

Comment (benvand) [reply job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Hey yeah, that's just the notify template id for our template. It's not possible to use it unless you have our account api key

Comment (Wynndow) [reply job_definitions/notify_suppliers_of_brief_withdrawals.yml]: Cool. Thumbs up from me then.

**Pull request merged**